### PR TITLE
Add support for adding a JMX interface to a host

### PR DESCRIPTION
**Pull request by Fabian1976**
#### Pull Request (PR) description
Add support for adding a JMX interface to host

Just add this to your hiera:
```
zabbix::agent::jmx_port: 9999
```

### CODE REVIEW

File: manifests/agent.pp [hunk 1]
```diff
@@ -282,6 +285,7 @@
   String $additional_service_params               = $zabbix::params::additional_service_params,
   String $service_type                            = $zabbix::params::service_type,
   Boolean $manage_startup_script                  = $zabbix::params::manage_startup_script,
+  Integer $jmx_port                               = $zabbix::params::agent_jmx_port,
```
Comment (bastelfreak) [inline manifests/agent.pp]: Can you use the [Stdlib::Port](https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/REFERENCE.md#stdlibport) datatype?
```suggestion
  Stdlib::Port $jmx_port                          = $zabbix::params::agent_jmx_port,
```

File: manifests/params.pp [hunk 2]
```diff
@@ -278,6 +278,7 @@
     $agent_logfile                          = '/var/log/zabbix/zabbix_agentd.log'
     $agent_logfilesize                      = '100'
   }
+  $agent_jmx_port                           = 0
```
Comment (bastelfreak) [inline manifests/params.pp]: Should the default be undef? In that case, the datatype should be `Optional[Stdlib::Port]`

Comment (Fabian1976) [reply manifests/params.pp]: I use the check on 0 later on in the code. So it should be 0

**Comment by Fabian1976** [comment]
What is the significance of the failed jobs? The puppet jobs seem to work. When will this be merged? Or is there something expected of me?

File: lib/puppet/type/zabbix_host.rb [hunk 3]
```diff
@@ -101,6 +101,10 @@ def insync?(is)
     desc 'Whether it is monitored by an proxy or not.'
   end
 
+  newproperty(:jmx_port) do
```
Comment (baurmatt) [inline lib/puppet/type/zabbix_host.rb]: Please add a data type validation.

File: lib/puppet/provider/zabbix_host/ruby.rb [hunk 4]
```diff
@@ -51,19 +51,34 @@ def create
...
+      {
+        type: 1,
+        main: 1,
+        ip: @resource[:ipaddress],
+        dns: @resource[:hostname],
+        port: @resource[:port],
+        useip: @resource[:use_ip] ? 1 : 0
+      }
+    ]
+    host_interfaces = if jmx_port != 0 && !jmx_port.nil?
```
Comment (baurmatt) [inline lib/puppet/provider/zabbix_host/ruby.rb]: Did you check if this works with `self.instances`?

File: manifests/params.pp (hunk 2 above)
Comment (baurmatt) [reply manifests/params.pp]: Please check for nil/undef. Checking for 0 seems kind like a hack :)

**Comment by baurmatt** [comment]
Hey @Fabian1976, thanks for this PR! :) I've left a couple of comments in the code, please address them. Regarding the failing tests: They look like they are related to your change. Could you please verify if your change actually works?

```
  Notice: Compiled catalog for ubuntu1404-64-1.c.travis-ci-prod-3.internal in environment production in 0.05 seconds
  Info: Applying configuration version '1572331677'
  Error: Could not set 'present' on ensure: Server answer API error
   {
    "code": -32602,
    "message": "Invalid params.",
    "data": "Incorrect arguments passed to function."
  }
   on request:
   {
    "method": "host.create",
    "params": [
      {
        "host": "test1.example.com",
        "interfaces": [
          {
            "type": 1,
            "main": 1,
            "ip": "127.0.0.1",
            "dns": "test1.example.com",
            "port": 10050,
            "useip": 1
          },
          {
            "type": 4,
            "main": 1,
            "ip": "127.0.0.1",
            "dns": "test1.example.com",
            "port": null,
            "useip": 1
          }
        ],
        "status": 0,
        "available": 1,
        "groups": [
          {
            "groupid": 8
          }
        ],
        "proxy_hostid": null,
        "templates": [
          {
            "templateid": 10001
          }
        ]
      }
    ],
    "id": 49721,
    "jsonrpc": "2.0",
    "auth": "0251d2407f003ca6b7bedfb02601f32f"
  } (file: /tmp/apply_manifest.pp.WYYzze, line: 1)
```

**Comment by Fabian1976** [comment]
mmm, the problem is. I wrote it for version 6.7.0 and tested it and it worked. Then i saw version 7.0.0 had a lot of major changes in the inner workings so i tried to rewrite it without testing. I will close my PR and submit it again once we've upgrade to puppet 6 (still on puppet 4) and can use version 7.0.0.

**Comment by fraenki** [comment]
AFAICT, the module still does not support adding JMX interfaces to a host. Is anybody else working on this? Otherwise I may try to resurrect the code from this PR...

**Pull request closed without merging**